I've removed 2.6.1 because of compilation error on Xcode
```
/Users/jenkins/workspace/cci_prod_PR-30134/conan-home/p/b/nlopt3462223af3297/b/src/src/algs/stogo/tools.h:39:24: error: no template named 'unary_function'; did you mean '__unary_function'?
   39 | class TrialGT : public unary_function<Trial, bool>
      |                        ^~~~~~~~~~~~~~
      |                        __unary_function
```
It can also be fixed by backporting https://github.com/stevengj/nlopt/commit/43f7081d9b83e04d25d729d51dc32e288008ca14#diff-f3fd494a1fd68a4b8f9484e6ccda8dd82fded632e876d68958fbd1e648d3988d

But there is probably no benefit in having 2.6.1 when there are newer versions available

--- a/recipes/nlopt/all/conandata.yml
+++ b/recipes/nlopt/all/conandata.yml
@@ -2,18 +2,3 @@ sources:
   "2.10.1":
     url: "https://github.com/stevengj/nlopt/archive/refs/tags/v2.10.1.tar.gz"
     sha256: "30d13ce16da119db3e987784f7864e35a562ec62c186352fae55cd003e6c58ff"
-  "2.10.0":
-    url: "https://github.com/stevengj/nlopt/archive/refs/tags/v2.10.0.tar.gz"
-    sha256: "506f83a9e778ad4f204446e99509cb2bdf5539de8beccc260a014bd560237be1"
-  "2.9.1":
-    url: "https://github.com/stevengj/nlopt/archive/refs/tags/v2.9.1.tar.gz"
-    sha256: "1e6c33f8cbdc4138d525f3326c231f14ed50d99345561e85285638c49b64ee93"
-  "2.7.1":
-    url: "https://github.com/stevengj/nlopt/archive/refs/tags/v2.7.1.tar.gz"
-    sha256: "db88232fa5cef0ff6e39943fc63ab6074208831dc0031cf1545f6ecd31ae2a1a"
-  "2.7.0":
-    url: "https://github.com/stevengj/nlopt/archive/v2.7.0.tar.gz"
-    sha256: "b881cc2a5face5139f1c5a30caf26b7d3cb43d69d5e423c9d78392f99844499f"
-  "2.6.2":
-    url: "https://github.com/stevengj/nlopt/archive/v2.6.2.tar.gz"
-    sha256: "cfa5981736dd60d0109c534984c4e13c615314d3584cf1c392a155bfe1a3b17e"

--- a/recipes/nlopt/all/conanfile.py
+++ b/recipes/nlopt/all/conanfile.py
@@ -1,11 +1,10 @@
 from conan import ConanFile
 from conan.tools.build import stdcpp_library
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
-from conan.tools.files import copy, get, replace_in_file, rm, rmdir
-from conan.tools.scm import Version
+from conan.tools.files import copy, get, rm, rmdir
 import os
 
-required_conan_version = ">=1.54.0"
+required_conan_version = ">=2"
 
 
 class NloptConan(ConanFile):
@@ -36,8 +35,6 @@ class NloptConan(ConanFile):
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if Version(self.version) < "2.9.0":
-            del self.options.enable_luksan
 
     def configure(self):
         if self.options.shared:
@@ -45,15 +42,14 @@ class NloptConan(ConanFile):
         if not self.options.enable_cxx_routines:
             self.settings.rm_safe("compiler.cppstd")
             self.settings.rm_safe("compiler.libcxx")
-        if not self.options.get_safe("enable_luksan", True):
+        if not self.options.enable_luksan:
             self.license = "MIT"
 
     def layout(self):
         cmake_layout(self, src_folder="src")
 
     def build_requirements(self):
-        if Version(self.version) >= "2.10.1":
-            self.tool_requires("cmake/[>=3.18]")
+        self.tool_requires("cmake/[>=3.18]")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
@@ -66,24 +62,14 @@ class NloptConan(ConanFile):
         tc.variables["NLOPT_OCTAVE"] = False
         tc.variables["NLOPT_MATLAB"] = False
         tc.variables["NLOPT_GUILE"] = False
-        if Version(self.version) >= "2.10.0":
-            tc.variables["NLOPT_JAVA"] = False
+        tc.variables["NLOPT_JAVA"] = False
         tc.variables["NLOPT_SWIG"] = False
-        if Version(self.version) >= "2.9.0":
-            tc.variables["NLOPT_LUKSAN"] = self.options.enable_luksan
+        tc.variables["NLOPT_LUKSAN"] = self.options.enable_luksan
         tc.variables["NLOPT_TESTS"] = False
         tc.variables["WITH_THREADLOCAL"] = True
         tc.generate()
 
-    def _patch_sources(self):
-        # don't force PIC
-        if Version(self.version) < Version("2.8.0"):
-            cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
-            replace_in_file(self, cmakelists, "set (CMAKE_C_FLAGS \"-fPIC ${CMAKE_C_FLAGS}\")", "")
-            replace_in_file(self, cmakelists, "set (CMAKE_CXX_FLAGS \"-fPIC ${CMAKE_CXX_FLAGS}\")", "")
-
     def build(self):
-        self._patch_sources()
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
@@ -100,7 +86,7 @@ class NloptConan(ConanFile):
             {"subdir": "slsqp" , "license_name": "COPYRIGHT"},
             {"subdir": "stogo" , "license_name": "COPYRIGHT"},
         ]
-        if self.options.get_safe("enable_luksan", True):
+        if self.options.enable_luksan:
             algs_licenses.append({"subdir": "luksan", "license_name": "COPYRIGHT"})
         for alg_license in algs_licenses:
             copy(self, alg_license["license_name"],
@@ -118,10 +104,6 @@ class NloptConan(ConanFile):
         self.cpp_info.set_property("cmake_target_name", "NLopt::nlopt")
         self.cpp_info.set_property("pkg_config_name", "nlopt")
 
-        self.cpp_info.names["cmake_find_package"] = "NLopt"
-        self.cpp_info.names["cmake_find_package_multi"] = "NLopt"
-        self.cpp_info.components["nloptlib"].names["cmake_find_package"] = "nlopt"
-        self.cpp_info.components["nloptlib"].names["cmake_find_package_multi"] = "nlopt"
         self.cpp_info.components["nloptlib"].set_property("cmake_target_name", "NLopt::nlopt")
         self.cpp_info.components["nloptlib"].set_property("pkg_config_name", "nlopt")
 

--- a/recipes/nlopt/config.yml
+++ b/recipes/nlopt/config.yml
@@ -1,13 +1,3 @@
 versions:
   "2.10.1":
     folder: all
-  "2.10.0":
-    folder: all
-  "2.9.1":
-    folder: all
-  "2.7.1":
-    folder: all
-  "2.7.0":
-    folder: all
-  "2.6.2":
-    folder: all

